main.py

```diff
@@ -17,9 +38,13 @@ def main():
 
     try:
         lines = read_scores(file_path)
+        summary_text = generate_summary_text(lines)
         txt_writer = TxtWriter()
         pdf_writer = PdfWriter()
-        general_summary(lines, output_path, save_as_txt, save_as_pdf, display_summary, txt_writer, pdf_writer)
+        terminal_printer = TerminalPrinter()
```

Comment:
Not all instances of writer classes will be needed every time. Therefore, they should be created only when needed:

In `main()`, create instances of writers depending on boolean flags. Then, pass a list of writer instances to `save_summary` function (together with the summary string).